### PR TITLE
Refactor DesignerActionPanel

### PR DESCRIPTION
**Pull request by halgab**
As discussed in #9422 @dreddy-work 

## Proposed changes

- Refactor `DesignerActionPanel` `Line` and `LineInfo` objects to increase the number of immutable fields in `Line` objects. The goal is to have a design that works better with NRT annotations

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10035)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs [hunk 1]
```diff
@@ -66,5 +62,15 @@ protected override void OnValueChanged()
         {
             _checkBox.Checked = (bool)Value;
         }
+
+        public sealed class Info(DesignerActionList list, DesignerActionPropertyItem item) : PropertyLineInfo(list, item)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs:66]: Please move nested class into a dedicated file

Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs:66]: I don't think that's a good idea here. I've designed the refactoring in such a way that each `Line` object has its own nested LineInfo class, and putting those classes in separate files will make us introduce several new files while degrading the readability of the code.
Of course, if you insist we should do it, I'll make the change. I'll just wait for you to confirm

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs:66]: The reason I suggested it was that the abstract base class, LineInfo, is in a dedicated file already and that seemed inconsistent to me. @dreddy-work  - what is your vote?

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.CheckBoxPropertyLine.cs:66]: @halgab - ok, let's keep your change as is.

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by lonitra — APPROVED** [review]

**Pull request merged**